lib/services/auth/auth_service.dart

```diff
@@ -0,0 +1,129 @@
+import 'dart:convert';
+
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:survey/gen/configs.gen.dart';
+import 'package:survey/services/locator/locator_service.dart';
+import 'package:survey/models/auth_token_info.dart';
+import 'package:object_mapper/object_mapper.dart';
+import 'package:survey/models/user_info.dart';
+import 'package:survey/services/api/api_service.dart';
+import 'package:survey/services/http/http_service.dart';
+import 'package:survey/services/user/user_service.dart';
+
+part 'params/auth_login_params.dart';
+
+abstract class AuthService {
+  static const loginEndPoint = "/oauth/token";
+  static const logoutEndPoint = "/oauth/revoke";
+  static const preferenceTokenKey = "auth_service_preference_token";
+
+  bool get isAuthenticated;
+
+  String? get token;
+
+  UserInfo? get user;
+
+  Future<void> login({
+    required AuthLoginParams params,
+  });
+
+  Future<void> logout();
+
+  Future<void> attempt();
+
+  Future<void> fetchUser();
```

Comment:
Authentication is different from Authorization - so if you plan to manage each service separately, then Authentication which is dealing with the token request, session sign out...then it should not involve the Profile data. 
I noticed that you're calling another Service from this Service, this is quite problematic imo...

If your idea is to manage the login state, current user's profile at 1 place, then I'd recommend you to create another `UserManager` class where you can depend on these AuthService and UserService. They should take a single responsibility and be consistent as the others from the same layer, i.e if all are Service, then they have the same communication to the remote backend.